inference-engine/src/auto_plugin/auto_plugin.cpp

```diff
@@ -274,34 +274,78 @@ DeviceName AutoInferencePlugin::SelectDevice(const std::vector<DeviceName>& meta
     }
 
     std::vector<DeviceName> CPU;
-    std::vector<DeviceName> GPU;
+    std::vector<DeviceName> dGPU;
+    std::vector<DeviceName> iGPU;
+    std::vector<DeviceName> MYRIAD;
+    std::vector<DeviceName> VPUX;
 
     for (auto& item : metaDevices) {
         if (item.find("CPU") == 0) {
             CPU.push_back(item);
             continue;
         }
-        if (item.find("GPU") == 0) {
-            GPU.push_back(item);
+        if (item.find("GPU.1") == 0) {
+            // dGPU found
+            dGPU.push_back(item);
+            continue;
+        } else if (item.find("GPU.0") == 0 || item.find("GPU") == 0) {
+            // iGPU found
```

Comment:
this GPU.0 or GPU also may be dGPU

Reply:
Will check device full name to update GPU device list: `Intel(R) Iris(R) Xe Graphics (iGPU)` and `Intel(R) Iris(R) Xe MAX Graphics (dGPU)` for example;